internal/otel/providers/external/external.go

```diff
@@ -15,15 +16,35 @@ import (
 
 type externalProvider struct {
 	exporterConfig *config.ExporterConfig
+	batchTimeout   time.Duration
+	metricsPort    int
+	envoyPort      int
+}
+
+type providerOpts func(*externalProvider)
+
+// WithTestOpts allows us to thread testing functionality into the config builder
+func WithTestOpts(metricsPort, envoyPort int, batchTimeout time.Duration) providerOpts {
```

Comment:
We should maybe call this something else, although it as added for the purposes of testing easily, it's behaviour is not for testing purposes, it just sets / mutates some values based on config object, and is used in the main code flow

Reply:
Should we carry it in as an unexported variable instead?

Something like ```
type CollectorCfg struct {
	ClientID          string
	ClientSecret      string
	ResourceID        string
	Client            hcp.TelemetryClient
	ForwarderEndpoint string
	ExporterConfig    *config.ExporterConfig
	opts []*otel.ProviderOpts?
}
```